artiq/master/worker_db.py

```diff
@@ -118,38 +115,34 @@ def __init__(self, ddb):
         self.ddb = ddb
         self._broadcaster.publish = ddb.update
 
-    def set(self, key, value, broadcast=False, persist=False, archive=True,
-            hdf5_options=None):
+    def set(self, key, value, broadcast=False, persist=False, archive=True):
+        if key in self.archive:
```

Comment:
This inadvertently reverts f42bea06a89ef80de4dcd4d996544a97287c969f and needs fixing before merge